### ISSUE TITLE
Microphone only works if I turn on speakerphone

### ISSUE THREAD

**Issue by JayGaring** [signalapp/Signal-iOS#2754]
- [X ] I have searched open and closed issues for duplicates

----------------------------------------

### Bug description
When making or receiving a call, my microphone only works when I press the speakerphone button. Not sure. but this might be related to issue 1669:

https://github.com/WhisperSystems/Signal-iOS/issues/1669

### Steps to reproduce
- Make or receive a call
- Other person can't hear me until I press speakerphone
- After I press Speakerphone, If I then turn off speakerphone, the other person can no longer hear me.

**Actual result:**  They can't hear me unless I enable the speakerphone
**Expected result:** They can always hear me regardless of the status of the speakerphone

### Screenshots
n/a

### Device info
**Device**:  iPhone 7 plus
**iOS version**: 11.1.
**Signal version:** 2.18.2.1

### Link to debug log
https://gist.github.com/c6fef92542bc186793bf00e69472334d

**Comment by michaelkirk** [comment]
Is this 100% reproducible? Or does it only seem to happen sometimes?

There are some changes in the upcoming 2.19 release which might help this.

We've got a 2.19 beta coming out soon. If you're willing to try unstable software, e.g. more likely to crash or lose data, you can get signed up by emailing support@whispersystems.org with the subject "Sign me up for the Signal iOS Beta"

**Comment by JayGaring** [comment]
Hi Michael,
It is consistent for me, even if I force close the app and re-launch it. I have to use the speakerphone all the time now. Glad to hear about the new release.

**Comment by michaelkirk** [comment]
@jaygaring would you be willing to try the latest beta? I’m curious to see if this will fix anything for you, as your situation sounds a little different than the one we just fixed.

Keep in mind that as a beta user you’d be using less stable software - it’s more likely to crash or lose data.

If you’re willing, please email support@whispersystems.org with the subject “sign me up for the Signal iOS beta”.

**Comment by JayGaring** [comment]
Sure, I'll try the beta. I'll send an email now.

**Comment by JayGaring** [comment]
I just updated to 2.19.0.19, but unfortunately it still has the same problem. The other person can only hear me if I turn on the speakerphone.

**Comment by michaelkirk** [comment]
Would you mind submitting a new debug log from 2.19?


Follow these steps if you can:

Call a friend
Verify they can’t here you
Toggle speakerphone, and verify they can here you.
Turn off speakerphone. Can they hear you now?


Then, as soon as the call ends, submit a debug log 

Thanks for your help getting this resolved.

**Comment by JayGaring** [comment]
Okay, here's a new debug log:
https://gist.github.com/677d0ef5d87aec4e7072fc960dfe7984 

I followed the steps above, and he problem is the same. Also, the other party said she could barely hear me when the speaker phone was off. However, I suspect that might have been true in the past, and it might not have been total silence in the past when I was not on speaker phone. I think this difference this time was because we tested it in a very quiet environment this morning.
